python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -287,6 +287,9 @@ def __reduce__(self) -> tuple:
         mapping = {name: getattr(self, name) for name in self.__slots__}
         return (_reconstructDimensionRecord, (self.definition, mapping))
 
+    def _ipython_display_(self) -> None:
```

Comment:
Is there a reason why this is being used and not `_repr_html_`?

Reply:
Yeah, I because I didn't want it to print in a different font or anything fancy at all, just to have it actually print the object correctly with all the nice newline formatting that y'all actually went to the trouble to include!

I totally agree on the docstring, but do you really think a test which is just calling `print()` (which just calls `str`) actually needs its own test?! Surely either `__str__` is already being tested, or it's not, and that was previously deemed OK, no?

Reply:
We need a test because how else do we know that the code isn't broken? Just because `__str__` is being tested doesn't mean that your new code doesn't have a problem. You are asserting by inspection that it must be fine.

As for `_repr_html_` wouldn't that be:

```python
def _repr_html_(self) -> str:
    return f"<pre>{self}<pre>"
```
?

Reply:
Sure, that looks fine. Would you prefer that for some reason?

Also, I'm not familiar with the tests in this package, and couldn't, at a glance, see the `test_records.py` or equivalent. There must be a place with an instantiated record object where I can test this easily - could you point me to it please?

Reply:
The `_repr_html_` version has the advantage of being more testable. Does it work?

How about a new method after this one: https://github.com/lsst/daf_butler/blob/main/tests/test_simpleButler.py#L599?

Reply:
Sure, looks good, thanks. Personally, creating a whole other butler just to test a print statement feels like a net-negative in terms of overkill and build-time slowdown, but it's your package, so I'm happy to include it.

Reply:
I think I've dealt with everything now, take a look and see what you think.

Reply:
I'm fine with you putting the ipython repr test in the JSON test loop.